installer/base.py

```diff
@@ -29,6 +30,18 @@ class InstallerError(Exception):
     pass
 
 
+# Type annotation not accepting Network maybe because the Network class is declared later
```

Comment:
It is better to separate the validation from the value-creating function. This way one can have a function that focus only on validating infura project ids, and another function that can generate the function in any format.